chain/connector/src/connector/accounts.rs

```diff
@@ -0,0 +1,104 @@
+use blokli_client::api::{BlokliQueryClient, BlokliTransactionClient};
+use futures::future::BoxFuture;
+use futures::{FutureExt, StreamExt};
+use futures::stream::BoxStream;
+use hopr_api::chain::{AccountSelector, AnnouncementError, ChainReceipt, Multiaddr};
+use hopr_crypto_types::prelude::*;
+use hopr_internal_types::account::AccountEntry;
+use hopr_primitive_types::prelude::*;
+
+use crate::connector::{Backend, HoprBlockchainConnector};
+use crate::errors::ConnectorError;
+use crate::payload::{sign_payload, PayloadGenerator};
+
+#[async_trait::async_trait]
+impl<B, C> hopr_api::chain::ChainReadAccountOperations for HoprBlockchainConnector<B, C>
+where
+    B: Backend + Send + Sync + 'static,
+    C: BlokliQueryClient + Send + Sync + 'static {
+    type Error = ConnectorError;
+
+    async fn node_balance<Cy: Currency>(&self) -> Result<Balance<Cy>, Self::Error> {
+        if Cy::is::<WxHOPR>() {
+            Ok(self.client.query_native_balance(&self.chain_key.public().to_address().into()).await?.balance.0.parse()?)
+        } else if Cy::is::<XDai>() {
+            Ok(self.client.query_token_balance(&self.chain_key.public().to_address().into()).await?.balance.0.parse()?)
+        } else {
+            Err(ConnectorError::InvalidState("unsupported currency"))
+        }
```

Comment:
Looks like this is a repeating code... that accepts only `self.chain_key.public()`, could be extracted to simplify also other calls.:
```suggestion
// extract this to a helper function
        if Cy::is::<WxHOPR>() {
            Ok(self.client.query_native_balance(&self.chain_key.public().to_address().into()).await?.balance.0.parse()?)
        } else if Cy::is::<XDai>() {
            Ok(self.client.query_token_balance(&self.chain_key.public().to_address().into()).await?.balance.0.parse()?)
        } else {
            Err(ConnectorError::InvalidState("unsupported currency"))
        }
```